### PR TITLE
[stylelint-config V2.0.0] Upgrade Dependencies

### PR DESCRIPTION
**Pull request by MoritzHayden**
## Description of Proposed Changes
- Upgrade `stylelint-config` dependencies
- Bump `stylelint-config` version to 2.0.0

## Related Issue (if applicable)
#31

### CODE REVIEW

File: packages/stylelint-config/package.json [hunk 1]
```diff
@@ -21,16 +21,18 @@
...
   "devDependencies": {
-    "jest": "^26.6.3"
+    "jest": "^27.5.1"
   },
   "dependencies": {
-    "stylelint-config-prettier": "^8.0.1",
-    "stylelint-config-sass-guidelines": "^7.1.0",
-    "stylelint-order": "^4.1.0"
+    "postcss": "^8.4.6",
+    "stylelint": "^14.3.0",
```
Comment (alexmacarthur) [inline packages/stylelint-config/package.json:32]: Do these need to be introduced as new dependencies? If so, why? I don't recall the reasoning, but I think we deliberately decided to only include `stylelint` as a `peerDepdencency.`

Comment (MoritzHayden) [reply packages/stylelint-config/package.json:32]: If I run `yarn install` without those dependencies, I get unmet peer dependency warnings. But now that I think about it, they would not be unmet if they are included in the project calling this library... is that right? If that's the case, I think `"postcss": "^8.4.6"` should also be included under peerDependencies, and then these two can be removed. Am I thinking about that correctly?

Without Dependencies

Comment (alexmacarthur) [reply packages/stylelint-config/package.json:32]: Yes! That's what I'd do -- remove them from `dependencies` and keep them in `peerDependencies`. That way, the consuming application has full control over the specific version of each they'd like to use.

Comment (MoritzHayden) [reply packages/stylelint-config/package.json]: Sounds good! Thank you for the input.

**Pull request merged**